pkg/api/validate/quota_test.go

```diff
@@ -0,0 +1,296 @@
+								Value: to.StringPtr("cores"),
+							},
+							CurrentValue: to.Int32Ptr(101),
+							Limit:        to.Int64Ptr(204),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("virtualMachines"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(113),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("standardDSv3Family"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(113),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("PremiumDiskCount"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(113),
+						},
+					}, nil)
+			},
+		},
+		{
+			name:    "not enough virtualMachines",
+			wantErr: "400: ResourceQuotaExceeded: .: Insufficient 'virtualMachines' quota: need 13, current available is 12 (101 of 113 used).",
+			mocks: func(tt *test, uc *mock_compute.MockUsageClient) {
+				uc.EXPECT().
+					List(ctx, "ocLocation").
+					Return([]compute.Usage{
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("cores"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(204),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("virtualMachines"),
+							},
+							CurrentValue: to.Int32Ptr(101),
+							Limit:        to.Int64Ptr(113),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("standardDSv3Family"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(113),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("PremiumDiskCount"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(113),
+						},
+					}, nil)
+			},
+		},
+		{
+			name:    "not enough standardDSv3Family",
+			wantErr: "400: ResourceQuotaExceeded: .: Insufficient 'standardDSv3Family' quota: need 13, current available is 12 (101 of 113 used).",
+			mocks: func(tt *test, uc *mock_compute.MockUsageClient) {
+				uc.EXPECT().
+					List(ctx, "ocLocation").
+					Return([]compute.Usage{
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("cores"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(204),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("virtualMachines"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(113),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("standardDSv3Family"),
+							},
+							CurrentValue: to.Int32Ptr(101),
+							Limit:        to.Int64Ptr(113),
+						},
+						{
+							Name: &compute.UsageName{
+								Value: to.StringPtr("PremiumDiskCount"),
+							},
+							CurrentValue: to.Int32Ptr(100),
+							Limit:        to.Int64Ptr(113),
+						},
+					}, nil)
+			},
+		},
```

Comment:
lots of tests like these are verbose and expensive to maintain.  You need at least one which checks for some insufficient quota case; the other insufficient quota cases you can infer by adding a separate table-based test for the behaviour of `addRequiredResources`.  Please sync with @m1kola.